tests/_test_process_executor.py

```diff
@@ -1085,6 +1085,25 @@ def check_viztracer_active_and_custom_init():
         finally:
             tracer.stop()
 
+    def test_exception_cause_with_tblib(self):
+        tblib_pickling_support = pytest.importorskip('tblib.pickling_support')
+
+        error_message = "This is the error message"
+
+        def raise_value_error():
+            tblib_pickling_support.install()
+            raise ValueError(error_message)
+
+        executor = self.executor_type(max_workers=2)
+        f = executor.submit(raise_value_error)
+        try:
+            f.result()
+        except ValueError as e:
+            assert e.__cause__ is not None
+            assert error_message in str(e.__cause__)
+
+        executor.shutdown(wait=True)
```

Comment:
At first, I though that the `pickling_support.install` would have border effects on the follow up tests but as it is only installed in the worker, it should work like this.

Another solution to avoid the need to `tblib` dependency would be this one:
```python
import copyreg
from loky import ProcessPoolExecutor


def raise_value_error():
    raise ValueError("The message")

def exception_subclasses():
    to_visit = [BaseException]
    while to_visit:
        this = to_visit.pop()
        yield this
        to_visit += list(this.__subclasses__())


def unpickle(func, args):
    exc = func(*args)
    exc.__cause__ = None
    return exc


def pickle_ex(ex):
    rv = ex.__reduce_ex__(3)
    return (unpickle, rv[:2]) + rv[2:]


def get_dispatch():
    modified_dispatch = copyreg.dispatch_table.copy()
    for exc in exception_subclasses():
        modified_dispatch[exc] = pickle_ex
    return modified_dispatch


def test_exception_cause():
    pe = ProcessPoolExecutor(max_workers=1, result_reducers=get_dispatch())
    f = pe.submit(raise_value_error)
    try:
        f.result()
    except ValueError as e:
        assert e.__cause__ is not None
        print(e.__cause__)
    else:
        raise RuntimeError


if __name__ == "__main__":
    test_exception_cause()
```

but it is probably too complicated so let's keep the `tblib` one as it is a small dependency for testing and it do not have border effects.

Reply:
> At first, I though that the `pickling_support.install` would have border effects on the follow up tests but as it is only installed in the worker, it should work like this.

For completeness, and for some reason I have not investigated, the issue does not happen if `tblib.pickling_support.install()` is called in the main program. In other words the snippet from the top post does not create the issue anymore, maybe something has changed in `tblib` ...